pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2100,7 +2104,16 @@ static __always_inline int do_should_trace(event_data_t *data)
 
     if (config & FILTER_BIN_PATH_ENABLED) {
         bool filter_out = (config & FILTER_BIN_PATH_OUT) == FILTER_BIN_PATH_OUT;
-        if (!equality_filter_matches(filter_out, &binary_filter, proc_info->binary))
+        // lookup by binary path only
+        u32 *equality = bpf_map_lookup_elem(&binary_filter, proc_info->binary.path);
```

Comment:
Shouldn't there be a check for `binary_no_mnt` instead of doing two lookups?

Reply:
Maybe you meant to rename but didn't, i'll mark the field in a review

Reply:
If the field is supposed to be set as I thought it was, then we can use it to determine which lookup to do. 
But better is if we know how to set it we can default the mount ns itself to -1 and then the hashing will always match without caring for the namespace.

Reply:
It is meant to be always 0.
It is used for the lookup, when we give the key starting from the binary path and not from mnt_id of the binary. Then, padding with zero at the end with `binary_no_mnt` is used to look for matches with no mount namespace.

Reply:
I asked this to be documented (also had doubts when first read, then when reading the bpf code it was clear).